### PR TITLE
Implementation of the target selection page with table + few hot fixes

### PR DESCRIPTION
**Pull request by AJK-dev**
The good, a new functionality matching this branch:
* New table-based target selection page

The bad, updates not related to this functionality:
* Updated structure state assignment
* Added maximum AA conservation column for structural comparisons

The ugly, some fixes that crept in:
* Added missing purge functionality for pathways
* Solved Excel interaction download error

Next up: awaiting the Codacy analysis and then adding a commit improving the code style

### CODE REVIEW

**Review by esguerra — COMMENTED** [review]
Seems like codacy is a bit too strict concerning camel case for ECMAScript, say, for yadcf. Maybe there can be someway to tell codacy to skip that linting?

**Comment by AJK-dev** [comment]
> Seems like codacy is a bit too strict concerning camel case for ECMAScript, say, for yadcf. Maybe there can be someway to tell codacy to skip that linting?

Indeed, I've just adjusted the setting in Codacy + for few minor other things adjusted the config in the JS file

**Review by esguerra — APPROVED** [review]

File: common/templates/common/selection_table.html [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+        <span>
+          <button type="button" onclick="onlySelectedTargets(this);" class="btn btn-xs btn-primary" id="only_selected">Only selected</button>
+          <button type="button" onclick="clearFilters();" class="btn btn-xs btn-primary">Clear Filters</button>
+          <button type="button" onclick="clearTargetSelection();" class="btn btn-xs btn-primary">Clear Selection</button>
+          <span id="hidden_filter_container" class="hidden"></span>
+        </span>
+      </div>
+      <div class="col-md-4">
+        <span class="text-center">
+          <button type="text" id="selection_table_info" class="btn" disabled></button>
```
[the remark this answers is not in the document]
Comment (AJK-dev) [reply common/templates/common/selection_table.html:99]: Good catch, converted a input to button without cleaning up.

File: common/templates/common/targetselectiontable.html [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+{% endblock %}
+
+{% block table %}
+    {{ table_data | safe }}
+{% endblock %}
+
+{% block addon_js %}
+    <script type="text/javascript">
+        var type_of_selection = '{{ type_of_selection }}';
```
[the remark this answers is not in the document]
Comment (AJK-dev) [reply common/templates/common/targetselectiontable.html:50]: Actually I don't need them - so cleaned them up :-)

**Pull request merged**